### PR TITLE
[FLINT] add OpenBLAS32

### PR DESCRIPTION
**Pull request by tthsqe12**
submitting first what worked for me on my box with julia 1.6

### CODE REVIEW

**Comment by tthsqe12** [comment]
waiting to see what @thofma and @fingolfin think

**Comment by thofma** [comment]
You should test whether Oscar.jl works with this on julia 1.6-nightly on linux and macOS. Does it?

**Comment by tthsqe12** [comment]
Oscar fails but I don't think it is related to this.

File: F/FLINT/build_tarballs.jl [hunk 1]
```diff
@@ -52,6 +52,18 @@ for f in ${WORKSPACE}/srcdir/patches/*.patch; do
   atomic_patch -p1 ${f}
 done
 
+if [[ "$nbits" == "64" ]] && [[ "$target" != aarch64-* ]]; then
```
Comment (giordano) [inline F/FLINT/build_tarballs.jl:55]: Note that this won't work in Julia v1.7, where we do ILP64 for aarch64 platforms :disappointed:

Comment (thofma) [reply F/FLINT/build_tarballs.jl:55]: Is there an example on Yggdrasil where this is properly done?

Comment (giordano) [reply F/FLINT/build_tarballs.jl:55]: Ehm.... :see_no_evil:

Comment (giordano) [reply F/FLINT/build_tarballs.jl:55]: Elsewhere I suggested using OpenBLAS32, which has the twfold advantage of non requiring renaming all symbols and not having this inconvenient split 1.6/1.7

Comment (thofma) [reply F/FLINT/build_tarballs.jl:55]: @tthsqe12 Do you care that we will have 32-bit BlasInt? If not `OpenBLAS32` might be an option. See https://github.com/JuliaPackaging/Yggdrasil/pull/697

Comment (tthsqe12) [reply F/FLINT/build_tarballs.jl]: Seems to work here and it worked my box. 🤷

Comment (giordano) [reply F/FLINT/build_tarballs.jl]: Patches aren't needed anymore if you're targeting OpenBLAS32

**Comment by thofma** [comment]
Don't merge yet.

**Pull request merged**